garage/tf/policies/categorical_conv_policy_with_model.py

```diff
@@ -107,7 +107,8 @@ def __init__(self,
     def _initialize(self):
         state_input = tf.placeholder(tf.float32, shape=(None, ) + self.obs_dim)
 
-        with tf.variable_scope(self._variable_scope):
+        with tf.variable_scope(
```

Comment:
IMO this is a really confusing way to assign `self._variable_scope`

why not just do it in the constructor?

```python
self._variable_scope = tf.variable_scope(...)
```

Reply:
After some testing, I think `tf.VariableScope` is working fine. The pros of using `tf.VariableScope` is that it's pickleable and can be used for reentering.

Reply:
How about just store the name and everytime we do
```
with tf.variable_scope(self.name, reuse=True):
    ....
```
?